server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestNodesInfoAction.java

```diff
@@ -69,20 +69,28 @@ public String getName() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        final NodesInfoRequest nodesInfoRequest = prepareRequest(request);
+        settingsFilter.addFilterSettingParams(request);
+
+        return channel -> client.admin().cluster().nodesInfo(nodesInfoRequest, new NodesResponseRestListener<>(channel));
+    }
+
+    static NodesInfoRequest prepareRequest(final RestRequest request) {
         String[] nodeIds;
         Set<String> metrics;
 
         // special case like /_nodes/os (in this case os are metrics and not the nodeId)
         // still, /_nodes/_local (or any other node id) should work and be treated as usual
         // this means one must differentiate between allowed metrics and arbitrary node ids in the same place
         if (request.hasParam("nodeId") && !request.hasParam("metrics")) {
-            Set<String> metricsOrNodeIds = Strings.tokenizeByCommaToSet(request.param("nodeId", "_all"));
+            String nodeId = request.param("nodeId", "_all");
+            Set<String> metricsOrNodeIds = Strings.tokenizeByCommaToSet(nodeId);
             boolean isMetricsOnly = ALLOWED_METRICS.containsAll(metricsOrNodeIds);
             if (isMetricsOnly) {
                 nodeIds = new String[]{"_all"};
                 metrics = metricsOrNodeIds;
             } else {
-                nodeIds = metricsOrNodeIds.toArray(new String[]{});
+                nodeIds = Strings.splitStringByCommaToArray(nodeId);
```

Comment:
`splitStringByCommaToArray` does not trim whitespace I think whereas `tokenizeByCommaToSet` did . If we don't want to change the parsing, we should use `tokenizeToStringArray` here I think.